xpk.py

```diff
@@ -2192,18 +2420,83 @@ def get_main_container(args, system, docker_image, resource_type) -> str:
                   echo XPK Start: $(date) ; _sigterm() ( kill -SIGTERM $! 2>/dev/null;); trap _sigterm SIGTERM;{gsutil_test_command}({command}) & PID=$!; while kill -0 $PID 2>/dev/null; do sleep 5; done; wait $PID; EXIT_CODE=$? ; {xpk_internal_commands} echo XPK End: $(date); echo EXIT_CODE=$EXIT_CODE; exit $EXIT_CODE
                 resources:
                   limits:
-                    {resource_type}: {system.chips_per_vm}
+                    {resources}
+                volumeMounts:
+                {pw_volume_mounts}
   """
   return yaml.format(args=args,
                    system=system,
+                   image_pull_policy=pw.add_image_pull_policy_for_pw(args),
```

Comment:
Why are some of these functions pw specific and some not?

I like the idea of pathways_xpk_helpers.py but we should include get_env_container that does the same.

---

I think we should use pathways_xpk_helper.py to store the yaml configurations that are pathways specific. Keep the functions that are shared between xpk and pathways in xpk.

Yaml configurations like the kueue settings, workload creation command, pathways yaml file are unique to pathways and can go in the pathways_helpers.py fille. 

The functions like get_env_container/ add_image_pull_policy might fork more in the future for other applications or new xpk features so we should keep them in xpk.py or some separate file that is xpk_shared_helpers since it could be used outside of pathways. Happy to explain this more! Let me know if it makes sense

Reply:
get_env_container is also used by non-pw workloads - it returns args.env in that case. (It's a shared function, hence lives in xpk.)

Ohh yes, I am already in the process of moving Pathways YAML extensions to the helper file!

This makes sense! xpk is currently a huge file with lots of helpers, I am happy to address readability improvements in the next PR!